addon/components/license-picker/component.js

```diff
@@ -0,0 +1,78 @@
+import Ember from 'ember';
+import layout from './template';
+
+export default Ember.Component.extend({
+    layout,
+    store: Ember.inject.service(),
+    licensesAvailable: Ember.A(),
+    showBorder: true,
+    showYear: true,
+    showCopyrightHolders: true,
+    showOtherFields: Ember.observer('nodeLicense', 'nodeLicense.text', function() {
+        let text = this.get('nodeLicense.text');
+        if (!text) {
+            return;
+        }
+        this.set('showYear', text.indexOf('{{year}}') !== -1);
+        this.set('showCopyrightHolders', text.indexOf('{{copyrightHolders}}') !== -1);
+    }),
+    didReceiveAttrs() {
+        if (!this.get('licenses')) {
+            this.get('store').query('license', { 'page[size]': 20 }).then(ret => {
+                this.set('licensesAvailable', ret);
+            });
+        } else {
+            this.set('licensesAvailable', this.get('licenses'));
+        }
+        if (!this.get('currentValues.year')) {
+            let date = new Date();
+            this.set('year', String(date.getUTCFullYear()));
+        } else {
+            this.set('year', this.get('currentValues.year'));
+        }
+        if (this.get('currentValues.copyrightHolders')) {
+            this.set('copyrightHolders', this.get('currentValues.copyrightHolders'));
+        }
+    },
+    _setNodeLicense: Ember.observer('licensesAvailable', function() {
+        if (!this.get('currentValues.licenseType')) {
+            this.set('nodeLicense', this.get('licensesAvailable.firstObject'));
```

Comment:
That is because providers will have default licenses that need to be preselected. Here they would be the first in the array.